mkosi/__init__.py

```diff
@@ -388,6 +388,64 @@ def configure_os_release(context: Context) -> None:
         if ArtifactOutput.os_release in context.config.split_artifacts:
             shutil.copy(osrelease, context.staging / context.config.output_split_os_release)
 
+        if ArtifactOutput.metainfo in context.config.split_artifacts:
+            # https://www.freedesktop.org/software/appstream/docs/sect-Metadata-OS.html
+            metainfo = '<?xml version="1.0" encoding="UTF-8"?>\n'
+            metainfo += '<component type="operating-system">\n'
+            id = name = home_url = pretty_name = None
+            for line in osrelease.read_text().splitlines():
+                if line.startswith("ID="):
+                    id = line.split("=")[1].strip('"')
+                elif line.startswith("NAME="):
+                    name = line.split("=")[1].strip('"')
+                elif line.startswith("HOME_URL="):
+                    home_url = line.split("=")[1].strip('"')
+                elif line.startswith("PRETTY_NAME="):
+                    pretty_name = line.split("=")[1].strip('"')
+
+            # The <id> format is reverse DNS style with home_url + id
+            if home_url:
+                from urllib.parse import urlparse
+
+                url = urlparse(home_url)
+                if url.netloc:
+                    netloc = url.netloc.split(".")
+                    netloc.reverse()
+                    netloc.remove("www")
+                    id_field = ".".join(netloc)
+                    if id:
+                        id_field += f".{id}"
+                    metainfo += f"  <id>{id_field}</id>\n"
+            else:
+                metainfo += f"  <id>{id}</id>\n"
+            if name:
+                metainfo += f"  <name>{name}</name>\n"
+            elif context.config.image_id:
+                metainfo += f"  <name>{context.config.image_id}</name>\n"
+            metainfo += f"  <summary>{id} image built with mkosi</summary>\n"
```

Comment:
When `id` is `None` this would give "None image built with mkosi"?

Reply:
ID is a mandatory field in os-release

Reply:
No, the man page says it: "If not set, a default of "ID=linux" may be used." so not, it is not mandatory (or the man page is outdated. Either use `id or "linux"`, or add an assert